kratos/modeler/clean_up_problematic_triangles_modeler.cpp

```diff
@@ -0,0 +1,306 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/timer.h"
+#include "utilities/entities_utilities.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+#include "modeler/clean_up_problematic_triangles_modeler.h"
+
+namespace Kratos
+{
+
+void CleanUpProblematicTrianglesModeler::SetupModelPart()
+{
+    KRATOS_TRY;
+
+    Timer::Start("CleanUpProblematicTrianglesModeler");
+
+    // We clean the mesh
+    const std::string entity_type = mParameters["entity_type"].GetString();
+    const IndexType first_node_id = mParameters["first_node_id"].GetInt();
+    const IndexType first_element_id = mParameters["first_element_id"].GetInt();
+    const IndexType first_condition_id = mParameters["first_condition_id"].GetInt();
+    const double area_tolerance = mParameters["area_tolerance"].GetDouble();
+    CleanUpProblematicGeometriesInMesh(mrModelPart, entity_type, first_node_id, first_element_id, first_condition_id, area_tolerance);
+
+    Timer::Stop("CleanUpProblematicTrianglesModeler");
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void CleanUpProblematicTrianglesModeler::CleanUpProblematicGeometriesInMesh(
+    ModelPart& rThisModelPart,
+    const std::string& rEntityType,
+    const IndexType FirstNodeId,
+    const IndexType FirstElementId,
+    const IndexType FirstConditionId,
+    const double AreaTolerance
+    )
+{
+    if (rEntityType == "element") {
+        CleanUpProblematicGeometries<Element>(rThisModelPart, FirstNodeId, FirstElementId, AreaTolerance);
+    } else if (rEntityType == "condition") {
+        CleanUpProblematicGeometries<Condition>(rThisModelPart, FirstNodeId, FirstConditionId, AreaTolerance);
+    } else if (rEntityType == "geometry") { // TODO: Implement geometry cleanup
+        KRATOS_ERROR << "Geometry cleanup not yet supported. Sorry" << std::endl;
```

Comment:
Nice that we're already taking this into account. Just for me to know, as far as I understand it shouldn't be too complicated to extend what you did to geometries. Am I wrong?

Reply:
Indeed, most probably you can easily achieve it by doing a templated `GetInstance` which calls `GetGeometry()` in the condition/element case and returns the entity itself in the geometry one.

Reply:
The geometry container is completely different ..., so it would need much more work

Reply:
mmm... if you say it you're most probably right (I don't see it right now). Nevertheless, les leave it for the future.